packages/flutter_tools/test/general.shard/dart_plugin_test.dart

```diff
@@ -538,7 +538,7 @@ void main() {
 
         },
         throwsToolExit(
```

Comment:
while we're at it, can we add a test for the pubspec error?

Reply:
I also thought this until I realized that the implementation for the pubspec error doesn't work: the reason is that the covered case for this error is: when a native federated inline plugin implementation is used, it does not have to be mentioned as interface or plugin in the pubspec (except from `flutter:\n plugin:\n platforms:\n $myplatform`). This case does not occur in the dart only plugin case, therefore the flag `throwOnPluginPubspecError` can be enabled.

If I write a test, it covers a case, which actually should not be covered, as the same pubspec is used for plugins with mixed implementations (dart and inline native ones).
Also removing or changing a test then means a breaking change, which is not the case, as the feature was never enabled.

At least, that's my theory. I can post some tests here to validate that theory.

Reply:
I added a test via 70af56b9c36e49db8d55f760d2717a2d8da0ac30 and if enabling [throwing with pubspec error](https://github.com/flutter/flutter/commit/70af56b9c36e49db8d55f760d2717a2d8da0ac30#diff-2fb2f7fe1d44f737a3b8e0fefdf353e06196290b8461daf7e2b7c33cb615f6a5R142) (like for the other tests), it throws. Although it shouldn't, as it is a valid native plugin implementation. 

Therefore the whole [error paragraph](https://github.com/flutter/flutter/blob/f0bb9d57e687e8faa4ec25fc28b8af43cd2ec1c8/packages/flutter_tools/lib/src/flutter_plugins.dart#L1275) should be removed. Right now it throws, but then is catched in production, which acts like a `continue` statement, which would be the correct implementation.

I would prefer to do this step in a separate PR though.